src/sinks/aws_s3.rs

```diff
@@ -285,17 +285,17 @@ fn generate_key(
 
 fn encode_event(
     event: Event,
-    batch_time_format: &String,
+    key_prefix: &Template,
     encoding: &Option<Encoding>,
 ) -> Result<PartitionInnerBuffer<Vec<u8>, Bytes>, ()> {
-    let log = event.into_log();
-
-    let key = if let Some(ValueKind::Timestamp(ts)) = log.get(&event::TIMESTAMP) {
-        ts.format(batch_time_format).to_string()
-    } else {
-        Utc::now().format(batch_time_format).to_string()
-    };
+    let key = key_prefix.render_string(&event).map_err(|missing_keys| {
+        warn!(
+            message = "Keys do not exist on the event. Dropping event.",
+            keys = field::debug(missing_keys)
+        );
+    })?;
```

Comment:
This error will kill the whole sink task iirc. We prob want to do what we did with cloudwatch and use `with_flat_map`.

Reply:
Good catch, that is tricky. Would be nice to trait-ify some of these things before too long so we can make this type of thing harder to do on accident.